RHOAIENG-3245: add methods to execute and return queries from trustyai service

christinaexyou opened the pull request:
addresses [RHOAIENG-3245](https://issues.redhat.com/browse/RHOAIENG-3245) by making adding the following methods:
* converts json files to pandas dfs and vice versa
* uploads data to service and/or model
* get metric request with the option to schedule for reoccurring intervals
* retrieves metric data from prometheus for specific time interval

demo nb to follow shortly

ruivieira commented on src/trustyai/utils/extras/metrics_service.py:
```diff
@@ -0,0 +1,198 @@
+            )
+        if response.status_code == 200:
+            return response.text
+        print(f"Error {response.status_code}: {response.reason}")
```
Exception here, perhaps, instead of print.

christinaexyou commented on src/trustyai/utils/extras/metrics_service.py:
```diff
@@ -0,0 +1,198 @@
+        self.trusty_url = (
+            "https://"
+            + (
+                subprocess.check_output(
```
I think that would be helpful, esp if we include methods for getting the ns, listing the pods/routes, and etc.

ruivieira commented on src/trustyai/utils/server/server.py:
```diff
@@ -0,0 +1,29 @@
+    """
+
+    def __init__(self):
+        k8s_client = config.new_client_from_config()
```
`new_client_from_config()` is actually calling `load_kube_config()`, which reads the context locally. This will work from outside the cluster, but probably not from inside.

On the other hand, `load_incluster_config()` uses the pod's service account. Works from inside the cluster, but not from outside.

Perhaps we could have something like (untested):

```python
try:
        k8s_client = config.load_incluster_config()
except config.ConfigException:
         k8s_client = config.load_kube_config()
```

ruivieira commented on src/trustyai/utils/server/server.py:
```diff
@@ -0,0 +1,29 @@
+Server module
+"""
+# pylint: disable = import-error, too-few-public-methods
+from kubernetes import config, client
```
We could add here a check to see if the optional `trustyai[api]` dependencies are installed. An example is at https://github.com/trustyai-explainability/trustyai-explainability-python/blob/e40dc588c1264f26beff10ed94024b06864a8e68/src/trustyai/language/detoxify/tmarco.py#L8

christinaexyou commented:
changes made: 
* replaced `new_client_from_config` to  `load_incluster_config()`
* renamed `server/server.py` to `api/api.py` 
* updated `metrics_service.py`, `pyproject.toml` and `workflow.yml` to reflect name change
